libra/libra
# [swarm][config] Enable config generation for full node swarm

## Motivation

Changes are made to swarm config generation (`config/config_builder/src/swarm_config.rs`) to enable generation of configuration for nodes in a full node swarm.

Currently, full node config generation also requires a `Path` to the directory storing configuration of the upstream validator node. This is needed since the upstream node config also needs to be updated to allow the created full nodes to connect to it. If this config generation is done "manually", the upstream node will also need to be restarted to start accepting connections from the full nodes since we do not yet have the ability to dynamically update a node's configuration.

The `libra_swarm` binary is also augmented (with a `-w` flag) to allow creation of a full node swarm in addition to a validator swarm. `libra_swarm` takes care of restarting the appropriate validator node once full node configs have been generated.

To start a full node swarm, users can run:
`./target/debug/libra_swarm -n <NUM_NODES>  -w`

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

1. Run local validator and full node swarms
2. Commit txns through client connected to AC on validator swarm
3. Confirm that txns can be read through client connected to AC on full node swarm.

## Related PRs
#825
